subseas_flex_fcst/maproom.py

```diff
@@ -131,40 +131,63 @@ def write_mapTitle(startDate, leadTime, leadTimeOptions):
     return f'{targetPeriod} {CONFIG["variable"]} Forecast issued {startDate}'
 
 @APP.callback(
-    Output("cdf_graph", "figure"),
-    Output("pdf_graph", "figure"),
-    Output("map", "click_lat_lng"),
     Output("layers_group", "children"),
     Output("latInput", "value"),
     Output("lngInput", "value"),
     Input("submitLatLng","n_clicks"),
     Input("map", "click_lat_lng"),
-    Input("startDate","value"),
-    Input("leadTime","value"),
     State("latInput", "value"),
     State("lngInput", "value")
 )
-def local_plots(n_clicks, click_lat_lng, startDate, leadTime, latitude, longitude):
+def pick_location(n_clicks, click_lat_lng, latitude, longitude):
     # Reading
-
-    fcst_mu, fcst_var, obs, hcst = read_cptdataset(leadTime, startDate, y_transform=CONFIG["y_transform"])
-    if click_lat_lng is None: #Map was not clicked
-        if n_clicks == 0: #Button was not clicked (that's landing page)
-            lat = (fcst_mu.Y[0].values+fcst_mu.Y[-1].values)/2
-            lng = (fcst_mu.X[0].values+fcst_mu.X[-1].values)/2
-        else: #Button was clicked
+    filesNameList = glob.glob(f'{DATA_PATH}/{CONFIG["forecast_mu_filePattern"]}')
+    startDate = re.search("\w{3}-\w{1,2}-\w{4}",filesNameList[0])
+    startDate = datetime.strptime(startDate.group(),"%b-%d-%Y").strftime("%b-%-d-%Y")
+    fcst_mu = predictions.selFile(
+        DATA_PATH,
+        CONFIG["forecast_mu_filePattern"],
+        list(CONFIG["leads"])[0],
+        startDate
+    )
+    if dash.ctx.triggered_id == None:
+        lat = fcst_mu.Y[int(fcst_mu.Y.size/2)].values
+        lng = fcst_mu.X[int(fcst_mu.X.size/2)].values
+    else:
+        if dash.ctx.triggered_id == "map":
+            lat = click_lat_lng[0]
+            lng = click_lat_lng[1]
+        else:
             lat = latitude
             lng = longitude
-    else: #Map was clicked
-        lat = click_lat_lng[0]
-        lng = click_lat_lng[1]
+        half_res = (fcst_mu["X"][1] - fcst_mu["X"][0]) / 2
+        tol = np.sqrt(2 * np.square(half_res)).values
+        try:
+            nearest_grid = fcst_mu.sel(X=lng, Y=lat, method="nearest", tolerance=tol)
+            lat = nearest_grid.Y.values
+            lng = nearest_grid.X.values
+        except KeyError:
+            lat = lat
+            lng = lng
+    return dlf.Marker(position=[lat, lng]), lat, lng
```

Comment:
Yes, this (switching on dash.ctx.triggered_id) feels better to me.

Reducing the level of if/then/else nesting would make it a bit easier to read.
```
if dash.ctx.triggered_id == None:
    ...
elif dash.ctx.triggered_id == "map":
    ...
else:
    ...
```

Reply:
hmmm... maybe... If trigger is None, I don't do snapping because it's already done by picking the default central point.

Reply:
Oh, yeah. You could do the snapping in all cases, just to make it simpler (it does no harm to re-snap a pre-snapped location), but leaving it as it is would also be ok with me.